src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -259,7 +259,7 @@ public void clearAndSelect(BibEntry bibEntry) {
         } else {
             // select new entry
             getSelectionModel().clearSelection();
-            findEntry(bibEntry).ifPresent(entry -> {
+            model.getViewModelByIndex(database.getDatabase().indexOf(bibEntry)).ifPresent(entry -> {
```

Comment:
Please move this to `private Optional<BibEntryTableViewModel> findEntry(BibEntry entry)`.

In that method, store the result `database.getDatabase().indexOf(bibEntry)`. If this optional is empty, issue a LOGGER.warn.  -- I think, this branch of execution is never touched if there is no error inside JabRef.